### PR TITLE
3.0 clean up config

### PR DESCRIPTION
**Pull request by benbc**

### CODE REVIEW

**Comment by benbc** [comment]
@jakewins Could you review this, please?

**Comment by MishaDemianenko** [comment]
@benbc this change will most probably brake all our dependent projects and related builds since empty config was quite popular, so will be super nice if you will be ready for that :)

File: community/kernel/src/main/java/org/neo4j/kernel/configuration/Config.java [hunk 1]
```diff
@@ -87,24 +90,24 @@ public Config( Map<String, String> inputParams, Class<?>... settingsClasses )
...
     }
 
-    /** Add more settings classes. */
-    public Config registerSettingsClasses( Iterable<Class<?>> settingsClasses )
+    /**
+     * Returns a copy of this config with the given modifications.
+     *
+     * @return a new modified config, leaves this config unchanged.
+     */
+    public Config with( Map<String, String> additionalConfig, Class<?>... settingsClasses )
```
Comment (jakewins) [inline community/kernel/src/main/java/org/neo4j/kernel/configuration/Config.java:106]: Nice

**Comment by jakewins** [comment]
Looks dope.

**Comment by benbc** [comment]
Thanks for the warning @MishaDemianenko.

**Pull request merged**